contracts/market/MarketsRegistry.sol

```diff
@@ -331,24 +342,59 @@ contract MarketsRegistry is OwnableUpgradeSafe, Proxiable, IMarketsRegistry {
         emit MarketDestroyed(address(market));
     }
 
+    function addFeeReceiver(
+        address receiver,
+        address _secondaryAddress,
+        uint8 _vaultPercentage
+    ) public onlyOwner {
+        require(receiver != address(0x0), "Invalid fee receiver address");
+        require(_secondaryAddress != address(0x0), "Invalid secondary address");
+
+        feeReceivers[receiver] = Receiver({
+            secondaryAddress: _secondaryAddress,
+            vaultPercentage: _vaultPercentage,
+            authorized: true
+        });
+    }
+
     /**
      * Allow owner to move tokens from the registry
      */
     function recoverTokens(IERC20 token, address destination)
         public
         override
-        onlyOwner
+        // onlyOwner
     {
         require(destination != address(0x0), "Invalid destination");
-
+        require(
+            feeReceivers[msg.sender].authorized 
```

Comment:
You are reading from storage 5 times in this function, It will be cheaper to add memory variable for feeReceivers[msg.sender] and use this var